internal/controller/isbservicerollout/isbservicerollout_controller.go

```diff
@@ -640,26 +638,133 @@ func (r *ISBServiceRolloutReconciler) ErrorHandler(isbServiceRollout *apiv1.ISBS
+
+}
+
+// Implemented functions for the progressiveController interface:
+
+// CreateUpgradingChildDefinition creates an InterstepBufferService in an "upgrading" state with the given name
+func (r *ISBServiceRolloutReconciler) CreateUpgradingChildDefinition(ctx context.Context, rolloutObject ctlrcommon.RolloutObject, name string) (*unstructured.Unstructured, error) {
+	isbsvcRollout := rolloutObject.(*apiv1.ISBServiceRollout)
+	metadata, err := getBaseISBSVCMetadata(isbsvcRollout)
+	if err != nil {
+		return nil, err
+	}
+	isbsvc, err := r.makeISBServiceDefinition(isbsvcRollout, name, metadata)
+	if err != nil {
+		return nil, err
+	}
+
+	labels := isbsvc.GetLabels()
+	labels[common.LabelKeyUpgradeState] = string(common.LabelValueUpgradeInProgress)
+	isbsvc.SetLabels(labels)
+
+	return isbsvc, nil
+}
+
+func (r *ISBServiceRolloutReconciler) getCurrentChildCount(rolloutObject ctlrcommon.RolloutObject) (int32, bool) {
+	isbsvcRollout := rolloutObject.(*apiv1.ISBServiceRollout)
+	if isbsvcRollout.Status.NameCount == nil {
+		return int32(0), false
+	} else {
+		return *isbsvcRollout.Status.NameCount, true
+	}
+}
+
+func (r *ISBServiceRolloutReconciler) updateCurrentChildCount(ctx context.Context, rolloutObject ctlrcommon.RolloutObject, nameCount int32) error {
+	isbsvcRollout := rolloutObject.(*apiv1.ISBServiceRollout)
+	isbsvcRollout.Status.NameCount = &nameCount
+	return r.updateISBServiceRolloutStatus(ctx, isbsvcRollout)
+}
+
+// IncrementChildCount updates the count of children for the Resource in Kubernetes and returns the index that should be used for the next child
+func (r *ISBServiceRolloutReconciler) IncrementChildCount(ctx context.Context, rolloutObject ctlrcommon.RolloutObject) (int32, error) {
+	currentNameCount, found := r.getCurrentChildCount(rolloutObject)
+	if !found {
+		currentNameCount = int32(0)
+		err := r.updateCurrentChildCount(ctx, rolloutObject, int32(0))
+		if err != nil {
+			return int32(0), err
+		}
+	}
+
+	err := r.updateCurrentChildCount(ctx, rolloutObject, currentNameCount+1)
+	if err != nil {
+		return int32(0), err
+	}
+	return currentNameCount, nil
+}
+
+// Recycle deletes child
+func (r *ISBServiceRolloutReconciler) Recycle(ctx context.Context, childObject *unstructured.Unstructured, c client.Client) error {
+	// TODO: Implement
+	return nil
+}
+
+// ChildNeedsUpdating determines if the difference between the current child definition and the desired child definition requires an update
+func (r *ISBServiceRolloutReconciler) ChildNeedsUpdating(ctx context.Context, existingChild, newChildDefinition *unstructured.Unstructured) (bool, error) {
+	// TODO: Implement
+	return false, nil
+}
+
+// take the existing ISBService and merge anything needed from the new ISBService definition
+func (r *ISBServiceRolloutReconciler) Merge(existingISBService, newISBService *unstructured.Unstructured) (*unstructured.Unstructured, error) {
```

Comment:
Why does `Merge()` need to be exported?

Reply:
Oh, I originally was calling it from outside of these Rollout reconcilers, but it looks like I'm not anymore, so I'll remove it from the interface and make them all lower case.